AdvLoggerPkg/UnitTests/LineParser/LineParserTestApp.c

```diff
@@ -564,15 +566,18 @@ BasicTests (
   UT_LOG_INFO ("\nExpected Length=%d\n", AsciiStrLen (Btc->ExpectedLine));
   UT_LOG_INFO ("\n = %a =\n", Btc->ExpectedLine);
 
-  if (mMessageEntry.MessageLen != AsciiStrLen (Btc->ExpectedLine)) {
+  // if (mMessageEntry.MessageLen != AsciiStrLen (Btc->ExpectedLine)) {
     DUMP_HEX (DEBUG_ERROR, 0, mMessageEntry.Message, mMessageEntry.MessageLen, "Actual   - ");
     DUMP_HEX (DEBUG_ERROR, 0, Btc->ExpectedLine, AsciiStrLen (Btc->ExpectedLine), "Expected - ");
-  }
+  // }
 
   UT_ASSERT_EQUAL (mMessageEntry.MessageLen, AsciiStrLen (Btc->ExpectedLine));
 
   // The following also verifies that the string is NULL terminated.
-  UT_ASSERT_MEM_EQUAL (mMessageEntry.Message, Btc->ExpectedLine, mMessageEntry.MessageLen + sizeof (CHAR8));
+  UT_ASSERT_MEM_EQUAL (
+    &(mMessageEntry.Message[sizeof (ADV_LOG_TIME_STAMP_RESULT) - 1]),

```

Comment:
Since this can be statically computed: `sizeof (ADV_LOG_TIME_STAMP_RESULT) - 1`, I'd put that into a macro to reduce the verbosity here.

```c
#define ADV_LOG_TIME_STAMP_STR_LEN  (sizeof (ADV_LOG_TIME_STAMP_RESULT) - 1)
```